app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any, Optional, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length: int = 0
+        self.capacity: int = 8
+        self.hash_table: list[Optional[tuple]] = [None] * self.capacity
+
+    def _get_key_value(self, key: Hashable) -> Any:
+        index = hash(key) % self.capacity
+        original_index = index
+
+        while self.hash_table[index]:
+            if self.hash_table[index][0] == key:
+                return self.hash_table[index][2]
+            index = (index + 1) % self.capacity
+
+            if index == original_index:
+                break
+
+        return None
+
+    def _set_key_value(self, key: Hashable, value: Any) -> None:
+        self._check_extend_memory()
+
+        index = hash(key) % self.capacity
+        original_index = index
+
+        while self.hash_table[index]:
+            if self.hash_table[index][0] == key:
+                self.hash_table[index] = (key, hash(key), value)
+                return
+            index = (index + 1) % self.capacity
+
+            if index == original_index:
```

Comment:
This check is redundant. Resizing mechanism should prevent this situation from occurring.